tsdate/util.py

```diff
@@ -121,7 +124,104 @@ def preprocess_ts(tree_sequence, minimum_gap=1000000, trim_telomeres=True):
         assert tree_sequence.num_sites == tree_sequence_trimmed.num_sites
         return provenance.record_provenance(
                 tree_sequence_trimmed, "preprocess_ts", minimum_gap=minimum_gap,
-                trim_telomeres=trim_telomeres, delete_intervals=delete_intervals)
+                remove_telomeres=remove_telomeres, delete_intervals=delete_intervals)
     else:
-        logger.info("No gaps to trim")
+        logger.info("No gaps to remove")
         return tree_sequence
+
+
+def nodes_time(tree_sequence, unconstrained=True):
+    nodes_age = tree_sequence.tables.nodes.time[:]
+    if unconstrained:
+        metadata = tree_sequence.tables.nodes.metadata[:]
+        metadata_offset = tree_sequence.tables.nodes.metadata_offset[:]
+        for index, met in enumerate(tskit.unpack_bytes(metadata, metadata_offset)):
+            if index not in tree_sequence.samples():
+                try:
+                    nodes_age[index] = json.loads(met.decode())["mn"]
+                except (KeyError, json.decoder.JSONDecodeError):
+                    raise ValueError("Tree Sequence must be tsdated with the\
+                                      Inside-Outside Method. Use unconstrained=False if\
+                                      not.")
+    return nodes_age
+
+
+def sites_time(tree_sequence, unconstrained=True, mutation_age="child",
```

Comment:
The function name is a bit baffling here to a newbie, I think. Should we call it "sites_time_from_ts" or something? Also, I'm wondering if "unconstrained", "mutation_age", and "ignore_multiallelic" should be keyword only (but maybe I'm going a bit overboard with the keyword-only args)

Reply:
Sounds good